### PR TITLE
Solved bug in yen.jl

### PR DESCRIPTION
**Pull request by aurorarossi**
In order to fix the bug reported in the issue  #103 I add in Dijkstra algorithm the optional argument `maxdist`  that specifies the maximum path distance beyond which all path distances are assumed to be infinite.

### CODE REVIEW

File: src/shortestpaths/dijkstra.jl [hunk 1]
```diff
@@ -20,19 +20,21 @@ Return a [`Graphs.DijkstraState`](@ref) that contains various traversal informat
...
-`state.pathcounts` holds a vector, indexed by vertex, of the number of shortest paths from the source to that vertex. 
+`state.pathcounts` holds a vector, indexed by vertex, of the number of shortest paths from the source to that vertex.
 The path count of a source vertex is always `1.0`. The path count of an unreached vertex is always `0.0`.
 
-* `trackvertices=false`: If true, 
+* `trackvertices=false`: If true,
 
 `state.closest_vertices` holds a vector of all vertices in the graph ordered from closest to farthest.
 
+* `maxdist::Float64` (default: `Inf`) specifies the maximum path distance beyond which all path distances are assumed to be infinite (that is, they do not exist).
```
Comment (gdalle) [inline src/shortestpaths/dijkstra.jl:36]: Does it have to be a `Float64`? It seems like this could be replaced with `typemax(eltype(distmx))`?

Comment (aurorarossi) [reply src/shortestpaths/dijkstra.jl:36]: yes! I have just committed your suggestion. Thank you!

[the remark this answers is not in the document]
Comment (gdalle) [reply src/shortestpaths/dijkstra.jl:36]:
```suggestion
* `maxdist` (default: `typemax(T)`) specifies the maximum path distance beyond which all path distances are assumed to be infinite (that is, they do not exist).
```

File: src/shortestpaths/dijkstra.jl [hunk 2]
```diff
@@ -73,7 +75,9 @@ function dijkstra_shortest_paths(
     distmx::AbstractMatrix{T}=weights(g);
     allpaths=false,
     trackvertices=false,
-) where {T<:Real} where {U<:Integer}
+    maxdist::T=typemax(T)
```
Comment (gdalle) [inline src/shortestpaths/dijkstra.jl:78]: Type constraint is unnecessary here

**Review by gdalle — CHANGES_REQUESTED** [review]

**Review by gdalle — APPROVED** [review]

**Pull request merged**